src/Components/AdminTools/AddNewCustomer.tsx

```diff
@@ -3,30 +3,27 @@ import { InputText } from 'primereact/inputtext'
 import { Dropdown, DropdownChangeEvent } from 'primereact/dropdown'
 import { Button } from 'primereact/button'
 import InputComponent from '../CommonComponent/InputComponent'
-import { CityProps } from '../../Type/CommonType'
+import { Country, Role, State } from '../../Type/CommonType'
 import { CustomerAdminDataProps } from '../../Type/ComponentBasedType'
+import useMetaData from '../CommonComponent/MetaDataComponent'
 
 const AddCustomer: React.FC<CustomerAdminDataProps> = ({ customerData, editMode }) => {
   const [name, setName] = useState('')
   const [id, setId] = useState('')
   const [phone, setPhone] = useState('')
   const [email, setEmail] = useState('')
-  const [role, setRole] = useState<CityProps | undefined>(undefined)
   const [street, setStreet] = useState('')
   const [apt, setApt] = useState('')
-  const [state, setState] = useState<CityProps | undefined>(undefined)
-  const [country, setCountry] = useState<CityProps | undefined>(undefined)
   const [zipCode, setZipCode] = useState('')
+  const [role, setRole] = useState<Role | undefined>(undefined)
+  const [country, setCountry] = useState<Country | undefined>(undefined)
+  const [state, setState] = useState<State | undefined>(undefined)
   const [password, setPassword] = useState('')
   const [confirmPassword, setConfirmPassword] = useState('')
-
-  const cities: CityProps[] = [
-    { name: 'New York', code: 'NY' },
-    { name: 'Rome', code: 'RM' },
-    { name: 'London', code: 'LDN' },
-    { name: 'Istanbul', code: 'IST' },
-    { name: 'Paris', code: 'PRS' },
-  ]
+  const { getMetaData } = useMetaData()
+  const [rolesData, setRolesData] = useState<Role[] | undefined>(undefined)
```

Comment:
Redundant assignment .

In React ,  if you create a state variable using the useState hook but do not assign an initial value to it, the value of that state will be undefined by default.